examples/qt/lineplot.py

```diff
@@ -0,0 +1,317 @@
+
+import fastplotlib as fpl
+
+try:
+    from PyQt6.QtWidgets import QApplication, QWidget, QVBoxLayout, QMainWindow
+    from PyQt6.QtCore import QTimer, Qt
+
+except ImportError:
+    from PyQt5.QtWidgets import QApplication, QWidget, QVBoxLayout, QMainWindow
+    from PyQt5.QtCore import QTimer, Qt
+
+def rotate(angle, axis_x, axis_y, axis_z):
+    """
+    Quaternion representing rotation around the given axis by the given angle.
+    """
+    a2 = angle/2.0
+    c = cos(a2)
+    s = sin(a2)
+    return (axis_x * s, axis_y * s, axis_z * s, c)
+
+class FastPlotMain(QMainWindow):
+
+    MAJOR_TICKS = 5
+    MINOR_TICKS = 4
+    DATAPOINTS  = 5000                   # number of data points per line
+    INTERVAL    = 16                     # display refresh in milliseconds
+    WHITE       = (1.0, 1.0, 1.0, 1.0)
+    BLACK       = (0.0, 0.0, 0.0, 1.0)
+    RED         = (1.0, 0.0, 0.0, 1.0)
+    GREEN       = (0.0, 1.0, 0.0, 1.0)
+    BLUE        = (0.0, 0.0, 1.0, 1.0)
+    DARK_GRAY   = (0.2, 0.2, 0.2, 1.0)
+    LIGHT_GRAY  = (0.9, 0.9, 0.9, 1.0)
+
+    def __init__(self):
+        super().__init__()
+
+        # ─── Window ──────────────────────────────────────────────────────────────
+        
+        self.setWindowTitle("fastplotlib Line Plot Test")
+        self.resize(800, 600)
+
+        # ─── Figure & Subplot ────────────────────────────────────────────────────
+
+        self.fig = fpl.Figure(
+            (1, 1),
+            size=(800, 600),
+            names = "Line Plot",
+        )
+
+        # Subplot
+        self.ax = self.fig[0, 0]
+
+        # Turn on axi rulers and option grid
+        self.ax.axes.visible = True
+        self.ax.background_color = self.WHITE
+        if self.ax.axes.grids:
+            self.ax.axes.grids.xy.visible = True
+            self.ax.axes.grids.xy.color   = self.DARK_GRAY
+
+        # ─── Docks: Title & Axis Labels ──────────────────────────────────────────  
+
+        # Title
+        self.ax.docks["top"].size = 30
+        self.ax.docks["top"].add_text(
+            "Line Plots",
+            font_size=16,
+            face_color=(0, 0, 0, 1),
+            anchor="middle-center",
+            offset=(0, 0, 0),
+        )
+        self.ax.docks["top"].background_color = self.WHITE
+
+        # X label
+        self.ax.docks["bottom"].size = 30
+        self.ax.docks["bottom"].add_text(
+            "X",
+            font_size=16,
+            face_color=(0, 0, 0, 1),
+            anchor="middle-center",
+            offset=(0, 0, 0),
+        )
+        self.ax.docks["bottom"].background_color = self.WHITE
+
+        # Y label
+        q = rotate(pi/2.0, 0., 0., 1.)  # rotate 90 deg around z-axis
+        self.ax.docks["left"].size = 30
+        self.ax.docks["left"].add_text(
+            "Y",
+            font_size=16,
+            face_color=(0, 0, 0, 1),
+            anchor="middle-center",
+            offset=(0, 0, 0),
+            rotation=q,
+        )
```

Comment:
pygfx will have seen soon, let's wait until we have the proper implementation https://github.com/pygfx/pygfx/issues/739